### PR TITLE
Expose tsdb status in receiver

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit implements the `api/v1/status/tsdb` API in the Receiver.

Signed-off-by: Filip Petkovski <filip.petkovsky@gmail.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Implemented the `api/v1/status/tsdb` endpoint in the Receiver.

## Verification

Manual testing

### CODE REVIEW

File: pkg/api/status/v1.go
Comment (fpetkovski) [inline pkg/api/status/v1.go:38]: We could remove these structs if https://github.com/prometheus/prometheus/pull/10783 gets merged.

**Review by saswatamcode — COMMENTED** [review]
Thanks! 🚀

Just a question! 🙂

File: pkg/receive/multitsdb.go
Comment (saswatamcode) [reply pkg/receive/multitsdb.go:303]: Maybe documenting this in some way would be great!

File: pkg/api/status/v1.go
Comment (yeya24) [inline pkg/api/status/v1.go:24]: Let's group the imports

File: pkg/api/status/v1.go
Comment (yeya24) [inline pkg/api/status/v1.go:87]: So this API can only fetch the TSDB stats for a specific tenant. Admins cannot use this API to to see the global TSDB stats (cross tenant).

Comment (fpetkovski) [reply pkg/api/status/v1.go:87]: Yes, that's correct. We could extend the endpoint to provide a global view by traversing all tenants and merging the results.

File: pkg/api/status/v1.go
Comment (GiedriusS) [reply pkg/api/status/v1.go:53]: It was merged, we can re-use the struct now :+1:

Comment (fpetkovski) [reply pkg/api/status/v1.go:53]: Do I need to go get the latest prometheus version from master/main, or do we need to wait for a new point release? I'm not sure how Thanos manages dependencies.

Comment (yeya24) [reply pkg/api/status/v1.go:53]: We can use latest main directly.

File: pkg/api/status/v1.go
Comment (yeya24) [reply pkg/api/status/v1.go:86]: Sounds good. Let's have an issue to track it after this pr is merged.

File: pkg/api/status/v1.go
Comment (yeya24) [inline pkg/api/status/v1.go:82]: Other federated APIs like rules, targets, exemplars are exposed as gRPC APIs at the store, and federations are done at the Querier level.
For this API do we have any plan to implement the gRPC version of it and then aggregated at the Query level?

Comment (fpetkovski) [reply pkg/api/status/v1.go:82]: So we have this issue at the moment: https://github.com/thanos-io/thanos/issues/5395

Maybe we can discuss federation there, but I think it can be a good idea to do it. We just have to see how merging is going to work, and whether we want to merge the results in a single json object in the result, or have one object per store target. I think this is a question for getting tsdb stats for all tenants as well.

Comment (saswatamcode) [reply pkg/api/status/v1.go:82]: Regarding merging, I'm wondering how data replication for Receive will affect it. AFAIU, we use quorum-based logic for replication, so maybe merging this would lead to some inaccuracies in TSDB statistics for a tenant?

Comment (fpetkovski) [reply pkg/api/status/v1.go:82]: Yes agreed. The stats can get very inaccurate if we merge across replicas.

File: pkg/api/status/v1.go
Comment (yeya24) [reply pkg/api/status/v1.go:53]: Yes, we need to update Cortex or Cortex folks will update Thanos when they need to

Comment (fpetkovski) [reply pkg/api/status/v1.go:53]: One more thing, what is the reason for using github.com/akanshat/cortex instead github.com/cortexproject/cortex? Which one should actually be updated?

Comment (saswatamcode) [reply pkg/api/status/v1.go:53]: I think we use it for groupcache changes https://github.com/thanos-io/thanos/pull/4818. Maybe Thanos version in fork can be updated cc: @akanshat?

Comment (akanshat) [reply pkg/api/status/v1.go:53]: Yes, so while adding groupcache we had to make some changes in Thanos, which affected cortex imports just a little. Then we decided to update my fork of cortex and point Thanos to it, and once groupcache changes were merged we were going to create a PR to Cortex, to have those changes merged and point Thanos back to the original cortex.

Now [#4651](https://github.com/cortexproject/cortex/pull/4651) is open, it's been reviewed by @GiedriusS and @bwplotka but not by any cortex maintainer yet. I am not sure what the next step is. @GiedriusS would you like to help me out with this?

Comment (fpetkovski) [reply pkg/api/status/v1.go:53]: While waiting for that to get merged, should we maybe rebase the fork against the latest master in Cortex in order to get this PR unblocked? Otherwise, we will blocked whenever we try to update the prometheus/prometheus dependency.

File: pkg/receive/multitsdb.go
Comment (fpetkovski) [reply pkg/receive/multitsdb.go:303]: Docs added 👍

File: pkg/api/status/v1.go
Comment (bwplotka) [reply pkg/api/status/v1.go:53]: No need for relase

Comment (bwplotka) [reply pkg/api/status/v1.go:53]: We could rebase cortex fork.

**Review by bwplotka — APPROVED** [review]
I think this is solid. It is a good start, and we can go from here to federated merged view if needed. Also federated does not mean "merged". We could make an API that does not merge or merge partially or something. Many options. 

We can reuse struct later on too.

**Comment by bwplotka** [comment]
Unrelated but important issue? 
image

Or main got broken from last consistent hashing?

**Comment by fpetkovski** [comment]
Looks like the classic non-nil interface pointing to a nil object. Should be fixed now.

**Review by bwplotka — APPROVED** [review]

**Review by bwplotka — APPROVED** [review]
Thanks, LGTM! 👍🏽

**Pull request merged**
